ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -0,0 +1,218 @@
+from sqlalchemy.engine.base import Engine
+from sqlalchemy.engine.reflection import Inspector
+from sqlalchemy.inspection import inspect
+from sqlalchemy.orm import Session
+
+from metadata.generated.schema.entity.data.table import TableData
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.entity.services.databaseService import DatabaseService
+from metadata.generated.schema.entity.tags.tagCategory import Tag
+from metadata.generated.schema.metadataIngestion.databaseServiceMetadataPipeline import (
+    DatabaseServiceMetadataPipeline,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.ingestion.api.source import SourceStatus
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.ometa.utils import ometa_logger
+from metadata.ingestion.source.database.database_source import DatabaseSourceService
+from metadata.ingestion.source.database.dbt_souce import DBTSource
+from metadata.ingestion.source.database.sql_column_handler import SqlColumnHandler
+from metadata.utils.connections import (
+    create_and_bind_session,
+    get_connection,
+    test_connection,
+)
+
+logger = ometa_logger()
+
+
+@dataclass
+class SQLSourceStatus(SourceStatus):
+    """
+    Reports the source status after ingestion
+    """
+
+    success: List[str] = field(default_factory=list)
+    failures: List[str] = field(default_factory=list)
+    warnings: List[str] = field(default_factory=list)
+    filtered: List[str] = field(default_factory=list)
+
+    def scanned(self, record: str) -> None:
+        self.success.append(record)
+        logger.info(f"Table Scanned: {record}")
+
+    def filter(self, record: str, err: str) -> None:
+        self.filtered.append(record)
+        logger.warning(f"Filtered Table {record} due to {err}")
+
+
+class CommonDbSourceService(DBTSource, SqlColumnHandler, DatabaseSourceService):
+    def __init__(
+        self,
+        config: WorkflowSource,
+        metadata_config: OpenMetadataConnection,
+    ):
+        self.config = config
+        self.source_config: DatabaseServiceMetadataPipeline = (
+            self.config.sourceConfig.config
+        )
+        # It will be one of the Unions. We don't know the specific type here.
+        self.service_connection = self.config.serviceConnection.__root__.config
+        self.status = SQLSourceStatus()
+
+        self.metadata_config = metadata_config
+        self.metadata = OpenMetadata(metadata_config)
+
+        self.service = self.metadata.get_service_or_create(
+            entity=DatabaseService, config=config
+        )
+        self.engine: Engine = get_connection(self.service_connection)
+        self.test_connection()
+
+        self._session = None  # We will instantiate this just if needed
+        self._connection = None  # Lazy init as well
```

Comment:
Now I don't remember what was the point of `_connection`. We're eagerly picking up the engine so it might not be needed